docs/data/data-grid/server-side-data/lazy-loading.md

```diff
@@ -2,14 +2,111 @@
+
+The area which triggers the new request can be changed using `scrollEndThreshold`.
+
+The demo below shows the infinite loading mode. Page size is set to `15` and the mock server is configured to return a total of `100` rows. Once the response does not contain any new rows, the grid stops requesting new data.
+
+{{"demo": "ServerSideLazyLoadingInfinite.js", "bg": "inline"}}
+
+## Updating the loading mode
+
+The grid changes the loading mode dynamically if the total row count gets updated in any of the three ways described above.
+
+Based on the previous and the new value for the total row count, the following scenarios are possible:
+
+- **Unknown `rowCount` to known `rowCount`**: When the row count is set to a valid value from an unknown value, the Data Grid switches to the viewport loading mode. It checks the number of already fetched rows and adds skeleton rows to match the provided row count.
+
+- **Known `rowCount` to unknown `rowCount`**: If the row count is updated and set to `-1`, the Data Grid resets, fetches the first page, and sets itself in the infinite loading mode.
+
+- **Known `rowCount` greater than the actual row count**: This can happen either by reducing the value of the row count after more rows were already fetched or if the row count was unknown and the grid in the inifite loading mode already fetched more rows. In this case, the grid resets, fetches the first page and continues in one of the modes depending on the new value of the `rowCount`.
+
+:::warning
+`rowCount` is expected to be static. Changing its value can cause the grid to reset and the cache to be cleared which may lead to performance and UX degradation.
+:::
+
+The demo below serves more as a showcase of the behavior described above and is not representing something you would implement in a real-world scenario.
+
+{{"demo": "ServerSideLazyLoadingModeUpdate.js", "bg": "inline"}}
+
+## Nested lazy loading 🚧
+
 :::warning
 This feature isn't implemented yet. It's coming.
 
-👍 Upvote [issue #10857](https://github.com/mui/mui-x/issues/10857) if you want to see it land faster.
+👍 Upvote [issue #14527](https://github.com/mui/mui-x/issues/14527) if you want to see it land faster.
 
-Don't hesitate to leave a comment on the same issue to influence what gets built. Especially if you already have a use case for this component, or if you are facing a pain point with the [current solution](https://mui.com/x/react-data-grid/row-updates/#lazy-loading).
+Don't hesitate to leave a comment on the same issue to influence what gets built. Especially if you already have a use case for this feature, or if you are facing a pain point with your current solution.
 :::
+
+When completed, it would be possible to use `unstable_lazyLoading` flag in combination with [Tree data](/x/react-data-grid/server-side-data/tree-data/) and [Row grouping](/x/react-data-grid/server-side-data/row-grouping/).
```

Comment:
```suggestion
When completed, it will be possible to use the `unstable_lazyLoading` flag in combination with [tree data](/x/react-data-grid/server-side-data/tree-data/) and [row grouping](/x/react-data-grid/server-side-data/row-grouping/).
```